src/TorSharp/TorSharpArchitecture.cs

```diff
@@ -1,12 +1,18 @@
-﻿namespace Knapcode.TorSharp
+﻿using System;
+
+namespace Knapcode.TorSharp
 {
     /// <summary>
     /// CPU architectures that TorSharp can run on.
     /// </summary>
+    [Flags]
     public enum TorSharpArchitecture
     {
-        Unknown,
-        X86,
-        X64,
+        Unknown = 0,
+        X86 = 1,
+        X64 = 2,
+        Arm = 4,
```

Comment:
nitpick: Hmm ... this seems a bit too clever. `x86` seems more like Intel's x86 rather than a generic "32-bit" flag. So a line like `Arm | X86` reads strangely because AFAIK there is no such architecture where Intel x86 and Arm co-exist. Instead of leveraging flags for a generic "is ARM" check, could we just do `== Arm32 || == Arm64` and avoid the `[Flags]` change?

Reply:
I made this because otherwise it's too much if else everywhere.
Is flag change so big? I don't think there is someone who use int or serialization to initialize this.
```C#
if (Architecture == TorSharpArchitecture.Arm32 || Architecture == TorSharpArchitecture.Arm64) { }
```
Instead of
```C#
if (Architecture.HasFlag(TorSharpArchitecture.Arm)) { }
```

Reply:
I think my concern here is that the `TorSharpArchitecture.Arm` state is an invalid state to exist. If we consider the bitmasks, `00000100` is the `Arm` state. It's not a possible state in real life but it can be easily expressed in code with `TorSharpArchitecture.Arm`. Generally, enums are used to model state accurately and with the help of the compiler. This state should not exist. Additionally, a `[Flags]` enum should be used when multiple states should be expressable. At runtime, you will only have one architecture so having an enum where you can express x64, x86, and ARM simultaneously seems very strange to me. This is my own personal philosophy around enums.

I'd rather have an `IsArm()` extension method on the `TorSharpArchitecture` enum which does the `||` check.

Reply:
Is that better?

```C#
public static bool Contains<T>(this T value, string enumValue) where T : Enum
{
    return value.ToString().Contains(enumValue);
}
```
```C#
if (_settings.Architecture.Contains("Arm"))
{
    if (!_settings.AllowUnofficialSources)
        throw new TorSharpException($"{nameof(_settings.AllowUnofficialSources)} == false, there is no official tor distribution for {_settings.Architecture} platform.");

    downloadableFile = await GetLatestDownloadableArmFileAsync(_httpClient,
        BaseArmUrl,
        fileNamePatternAndFormat,
        CancellationToken.None).ConfigureAwait(false);
}
```
TorSharpArchitecture now
```C#
public enum TorSharpArchitecture
{
    Unknown,
    X86,
    X64,
    Arm32,
    Arm64
}
```

Reply:
Seems fine. The string allocation for `ToString()` is a bit odd but I doubt it will be a performance issue.